ikarus/io/resultfunction.hh

```diff
@@ -115,6 +116,48 @@ public:
         fes_{fes},
         userFunction_{UserFunction{}} {}
 
+  /**
+   * @brief Creates the ResultFunction as a function that can be used with dune-vtk
+   *
+   * Constructs a ResultFunction object with given finite elements, ferequirements as `Dune::Vtk::Function`
+   *
+   * @param fes Pointer to a vector of finite elements
+   * @param req FERequirements for evaluation
+   */
+  static auto asVtkFunction(std::vector<ElementType>* fes, const FERequirementType& req) {
+    return Dune::Vtk::Function<GridView>(std::make_shared<ResultFunction>(fes, req));
+  }
+
+  /**
+   * @brief Creates the ResultFunction as a Localfunction that can queried element-wise
+   *
+   * Constructs a ResultFunction object with given finite elements, ferequirements as `Dune::Vtk::LocalFunction`
+   * This leverages the Dune::VTK localfunction interface to be able to query element-wise
+   * Usage:
+   * @code
+   * auto localResultFunction = Ikarus::ResultFunction<FiniteElement, resType>::asLocalFunction(&fes,
+   * feRequirements); localResultFunction.bind(element); auto result = localResultFunction(pos);
+   * @endcode
+   * @param fes Pointer to a vector of finite elements
+   * @param req FERequirements for evaluation
+   */
+  static auto asLocalFunction(std::vector<ElementType>* fes, const FERequirementType& req) {
+    return localFunction(asVtkFunction(fes, req));
+  }
+
+  /**
+   * @brief Creates the ResultFunction as a function that can be used with dune-vtk
+   *
+   * Constructs a ResultFunction object with given finite elements, ferequirements as shared_ptr to be used with
+   * the native Dune VTKWriter
+   *
+   * @param fes Pointer to a vector of finite elements
+   * @param req FERequirements for evaluation
+   */
+  static auto asShared(std::vector<ElementType>* fes, const FERequirementType& req) {
```

Comment:
Creating this member function just to create a copy of yourself as shared_ptr is quite unusual. I would suggest removing this function.

You can think about creating a factory function that creates the ResultFunction and this would return by default a shared_ptr. Like in `makeNewtonRaphson`.

Reply:
Then you can pass this shared_ptr to VtkFunction or LocalFunction

Reply:
Then maybe at call site:

```cpp
auto resultFunction = makeResultFunction<resType,ResultEvalutor>(&fes,req); // auto=std::shared_ptr<ResultFunction>
auto vtkFunction = makeResultVtkFunction<resType,ResultEvalutor>(&fes,req); // auto=Dune::Vtk::Function
```

and then

```cpp
auto resultFunction = localFunction(makeResultVtkFunction<resType,ResultEvalutor>(&fes,req)); 
```